other/upgrade_2-1_mysql.sql

```diff
@@ -2066,6 +2066,14 @@ UPDATE {$db_prefix}personal_messages SET body = REPLACE(REPLACE(body, '[blue]',
 ---#
 
 /******************************************************************************/
+--- Tokenizing references to forum URL in posts and personal messages
+/******************************************************************************/
+---# Replacing $boardurl with token
+UPDATE {$db_prefix}messages SET body = REPLACE(body, '{$boardurl}', CONCAT('{$', 'boardurl}')) WHERE body LIKE '%{$boardurl}%';
+UPDATE {$db_prefix}personal_messages SET body = REPLACE(body, '{$boardurl}', CONCAT('{$', 'boardurl}')) WHERE body LIKE '%{$boardurl}%';
+---#
```

Comment:
Hope you realize what that sort of update this does on a multi-million post forum.  This will be a killer query.

Reply:
Not much different that all the BBC replacement queries immediately above it.

Reply:
wow those will also be killers.  Well I guess accept this PR as it is.  We should address this separately.  I couldn't upgrade sm.org with that.  A LIKE wildcard (%) on the messages table would kill the database to try and figure that out.  If it could use the sphinx search index, then it wouldn't be that bad.

Reply:
If you can come u p with a better method, I'd be 100% in favour of that.

Reply:
Well again, this would be outside of this PR.

But I think we could create a scheduled task, call it something like "Background Upgrade Tasks".  This would have the code to process this a set of messages at a time (such as lets say 100 of them).  We would track this upgrade process in a separate table from the settings table (to avoid causing a recreate of the settings cache for every update).  This would also be able to upgrade personal messages.

If we do it this way, the users would experience for a short while the possibility of the old bbc codes not working and the boardurl replacement not functioning on older posts.  Once the background upgrader went through, it would resolve it.

The best part is, this allows us to easily keep cleaning up parts of our database that are heavy hitters for large forums, while keeping the main upgrader faster.

This of course only works with things like messages and pms where it would be acceptable for old BBC to not work.  If we did it on other things, we may have to have combat code in our system to handle this gracefully.